tests/tensors/constexpr_symmetric_tensor.cc

```diff
@@ -111,6 +111,37 @@ main()
   DEAL_II_CONSTEXPR const auto B = get_tensor_4();
   deallog << "SymmetricTensor<4,2> = " << B << std::endl;
 
+  {
+    DEAL_II_CONSTEXPR double a_init[3][3] = {{1., 0., 0.},
+                                             {2., 1., 0.},
+                                             {3., 2., 1.}};
+    DEAL_II_CONSTEXPR Tensor<2, 3> dummy_a{a_init};
+    DEAL_II_CONSTEXPR SymmetricTensor<2, 3> a = symmetrize(dummy_a);
+    std::cout << a << std::endl;
+    DEAL_II_CONSTEXPR auto inverted = invert(a);
+    std::cout << inverted << std::endl;
+    DEAL_II_CONSTEXPR double ref_init[3][3] = {{0., -2., 2.},
+                                               {-2., 5., -2.},
+                                               {2., -2., 0.}};
+    DEAL_II_CONSTEXPR Tensor<2, 3> dummy_ref{ref_init};
+    DEAL_II_CONSTEXPR SymmetricTensor<2, 3> ref = symmetrize(dummy_ref);
+    std::cout << ref << std::endl;
+    Assert(inverted == ref, ExcInternalError());
+  }
+  {
+    DEAL_II_CONSTEXPR double a_init[3][3] = {{1., 2., 3.},
+                                             {2., 1., 2.},
+                                             {3., 2., 1.}};
+    DEAL_II_CONSTEXPR Tensor<2, 3> dummy_a{a_init};
+    DEAL_II_CONSTEXPR SymmetricTensor<2, 3> a          = symmetrize(dummy_a);
+    DEAL_II_CONSTEXPR auto                  transposed = transpose(a);
+    Assert(transposed == a, ExcInternalError());
+    constexpr auto dummy   = scalar_product(a, a);
```

Comment:
```suggestion
    DEAL_II_CONSTEXPR const auto dummy   = scalar_product(a, a);
```
We need to make sure these lines work within pre-c++14 compilers. For example, in c++11, `a` is not constexpr and this line fails. There are other similar lines that we should fix before merging this.